### PR TITLE
Fix #3354:[RTL] High-fi Interaction View 

### LINKED ISSUE #3354: [RTL] High-fi Interaction View
Number Input, TextInput, Fractions and Ratios Interaction View need to support RTL.
screenshot
screenshot
screenshot

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR fixes #3354 , This Pr fixes NumberInput, TextInput, Fractions and Ratios Interaction View.

## Screenshot  RTL
Mobile 

screenshot.......screenshot
screenshot..... screenshot

screenshot.......screenshot

screenshot........screenshot

screenshot.....screenshot.

screenshot.........screenshot

screenshot,,,, screenshot
screenshot......screenshot

Tablet
screenshot....... screenshot

screenshot..........screenshot

screenshot...........screenshot

screenshot.........screenshot

screenshot.......screenshot

screenshot.......screenshot

screenshot.......screenshot

screenshot..........screenshot

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@veena14cs Can you please update the description to correctly show the LTR vs RTL screenshots.

Also, why is hint not showing RTL behaviour? If its because of non-strings.xml line then in that case can you add hint to `NumericInput` via `strings.xml` and then use that produce LTR vs RTL screenshots.

**Comment by veena14cs** [comment]
> @veena14cs Can you please update the description to correctly show the LTR vs RTL screenshots.
> 
> Also, why is hint not showing RTL behaviour? If its because of non-strings.xml line then in that case can you add hint to `NumericInput` via `strings.xml` and then use that produce LTR vs RTL screenshots.

I have updated Screenshots for LTR and RTL. The hint texts in screenshot are hardcoded  in `strings.xml` for testing purpose, the actual values of hints are not changed in this PR.

File: app/src/main/res/layout/fraction_interaction_item.xml [hunk 1]
```diff
@@ -10,16 +10,22 @@
       name="viewModel"
       type="org.oppia.android.app.player.state.itemviewmodel.FractionInteractionViewModel" />
   </data>
+  <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (rt4914) [inline app/src/main/res/layout/fraction_interaction_item.xml:13]: Can we get rid of either the `ContraintLayout` or the `LinearLayout` ? Keeping both does not make sense.

File: app/src/main/res/layout/fraction_interaction_item.xml [hunk 2]
```diff
@@ -10,16 +10,22 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:paddingEnd="@{viewModel.hasConversationView ?@dimen/interaction_item_exploration_view_margin_end : @dimen/interaction_item_question_view_margin_end}"
```
Comment (rt4914) [inline app/src/main/res/layout/fraction_interaction_item.xml]: Add space after `?`
Example: `viewModel.hasConversationView ? @dimen/interaction_item_exploration_view_margin_end`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs PTAL

File: app/src/main/res/layout/fraction_interaction_item.xml (hunk 1 above)
Comment (veena14cs) [reply app/src/main/res/layout/fraction_interaction_item.xml:13]: Done here and to other interactions too.

File: app/src/main/res/layout/fraction_interaction_item.xml (hunk 2 above)
Comment (veena14cs) [reply app/src/main/res/layout/fraction_interaction_item.xml]: Done.

File: app/src/main/res/layout-land/drag_drop_interaction_item.xml [hunk 3]
```diff
@@ -32,7 +32,7 @@
     app:explorationViewMarginApplicable="@{viewModel.hasConversationView &amp;&amp; !viewModel.splitView}"
     app:explorationViewMarginBottom="@{@dimen/space_0dp}"
     app:explorationViewMarginEnd="@{@dimen/drag_drop_item_exploration_view_margin_end}"
-    app:explorationViewMarginStart="@{@dimen/drag_drop_interaction_item_exploration_view_margin_start}"
+    app:explorationViewMarginStart="@{@dimen/drag_drop_interaction_item_exploration_view_padding_start}"
```
Comment (rt4914) [inline app/src/main/res/layout-land/drag_drop_interaction_item.xml]: Please revert this change as this PR is not focused on `DragAndDrop`, also this PR does not exists anymore in develop and its name should be `drag_drop_interaction_item_exploration_view_margin_start ` instead of `drag_drop_interaction_item_exploration_view_padding_start `

File: app/src/main/res/values-land/dimens.xml [hunk 4]
```diff
@@ -16,7 +16,7 @@
   <dimen name="drag_drop_item_question_view_margin_end">64dp</dimen>
   <dimen name="drag_drop_item_question_view_margin_start">64dp</dimen>
 
-  <dimen name="drag_drop_interaction_item_exploration_view_margin_start">80dp</dimen>
+  <dimen name="drag_drop_interaction_item_exploration_view_padding_start">80dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Revert this change

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 5]
```diff
@@ -46,17 +46,17 @@
   <dimen name="submitted_answer_exploration_view_margin_start">208dp</dimen>
   <dimen name="text_input_interaction_item_conversation_view_margin_start">208dp</dimen>
 
-  <dimen name="continue_interaction_item_question_view_margin_end">192dp</dimen>
-  <dimen name="continue_interaction_item_question_view_margin_start">192dp</dimen>
+  <dimen name="continue_interaction_item_question_view_padding_end">192dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Why is this getting changed? This PR is not using this change anywhere.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 6]
```diff
@@ -46,17 +46,17 @@
...
   <dimen name="continue_navigation_item_question_view_margin_start">192dp</dimen>
   <dimen name="drag_drop_item_question_view_margin_end">192dp</dimen>
   <dimen name="drag_drop_item_question_view_margin_start">192dp</dimen>
   <dimen name="state_fragment_non_split_view_margin_end">192dp</dimen>
   <dimen name="state_fragment_non_split_view_margin_start">192dp</dimen>
 
-  <dimen name="continue_interaction_item_exploration_view_margin_end">176dp</dimen>
-  <dimen name="continue_interaction_item_exploration_view_margin_start">176dp</dimen>
+  <dimen name="continue_interaction_item_exploration_view_padding_end">176dp</dimen>
+  <dimen name="continue_interaction_item_exploration_view_padding_start">176dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Why is this getting changed? This PR is not using this change anywhere.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 7]
```diff
@@ -14,8 +14,8 @@
   <dimen name="state_fragment_split_view_margin_end">32dp</dimen>
   <dimen name="profile_chooser_description_margin_start_profile_not_added">32dp</dimen>
 
-  <dimen name="continue_interaction_item_question_view_margin_end">128dp</dimen>
-  <dimen name="continue_interaction_item_question_view_margin_start">128dp</dimen>
+  <dimen name="continue_interaction_item_question_view_padding_end">128dp</dimen>
+  <dimen name="continue_interaction_item_question_view_padding_start">128dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Why is this getting changed? This PR is not using this change anywhere.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 8]
```diff
@@ -58,8 +58,8 @@
   <dimen name="text_input_interaction_item_conversation_view_margin_start">144dp</dimen>
   <dimen name="profile_chooser_fragment_margin_top_profile_not_added">124dp</dimen>
 
-  <dimen name="continue_interaction_item_exploration_view_margin_end">112dp</dimen>
-  <dimen name="continue_interaction_item_exploration_view_margin_start">112dp</dimen>
+  <dimen name="continue_interaction_item_exploration_view_padding_end">112dp</dimen>
+  <dimen name="continue_interaction_item_exploration_view_padding_start">112dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Why is this getting changed? This PR is not using this change anywhere.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs Suggested nit changes.

File: app/src/main/res/layout-land/drag_drop_interaction_item.xml (hunk 3 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/drag_drop_interaction_item.xml]: Done.

File: app/src/main/res/values-land/dimens.xml (hunk 4 above)
Comment (veena14cs) [reply app/src/main/res/values-land/dimens.xml]: Done.

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 5 above)
Comment (veena14cs) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: Done.

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 6 above)
Comment (veena14cs) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: Done

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 7 above)
Comment (veena14cs) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: Done.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 8 above)
Comment (veena14cs) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: Done.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by rt4914** [comment]
@veena14cs Update with latest develop.
